### DISCUSSION TITLE
blog/recent-public-mqtt-broker-changes/

### DISCUSSION THREAD

**Discussion by giscus** [meshtastic/Discussions#17] [Blog Post Comments]
# blog/recent-public-mqtt-broker-changes/

Recent changes to the Meshtastic project-hosted MQTT server enhance user privacy and data protection.

https://meshtastic.org/blog/recent-public-mqtt-broker-changes/

**Comment by GaryP136** [comment]
Apart from obvious load reduction benefits I think this is a great idea for all our security of location and tracking data.  Good work you guys!

**Comment by cyberorg** [comment]
“Server-Level Filtering: Only position packets with imprecise location information will be passed to the topic, ensuring that sensitive data is not exposed. Note: This filtering is only applied to the default key.”

Hi, This will be plain confusing, HIGH_PRECISION if selected by user should be respected. SAR depends on accuracy of the position, server should not filters out those positions.

**Reply by garthvh** [reply]
This is only possible on the default known global channel key, which should not be used for SAR operations.

**Comment by cyberorg** [comment]
So, do we need to set up separate channel for position precision? And default key/unencrypted can’t be used there on that channel if we want precise location?

Does this apply to only mqtt.meshtastic.org or all mqtt?

**Reply by rcarteraz** [reply]
Yes, precise location filtering is only done on the default key (since it's known). Just so you know, the allowed range of precision is 10-16 bits. If set to 16 bits, that's ± 364m. That's still pretty close and you're going to get a very good understanding of where that node is at.

**Reply by cyberorg** [reply]
Does this apply to only mqtt.meshtastic.org or all mqtt?

I tried new channel with different key, text goes through but position doesn’t show up, nothing comes on our mqtt. Using firmware 2.4.3. 

So far it seems precise position is seen on longfast channel using default key. I hope this behaviour will not change for private mqtt.

**Reply by valzzu** [reply]
> Does this apply to only mqtt.meshtastic.org or all mqtt?
> 
> I tried new channel with different key, text goes through but position doesn’t show up, nothing comes on our mqtt. Using firmware 2.4.3.
> 
> So far it seems precise position is seen on longfast channel using default key. I hope this behaviour will not change for private mqtt.

Only on mqtt.meshtastic.org.
The reason for no position going on the private channel is that it wasnt primary right?
U can still request position on secondary but position is not sent there automatically as im aware

**Comment by mkgin** [comment]
Enabling MQTT map reports could reasonably **imply consent** for whatever level of precision you specify when configuring a node.

I was initially surprised to see nodes with precise locations or precision less than 16 bits appear on meshmap.net but decided to read more carefully what was said regarding the changes to MQTT.

The blog post specifies:  *filtering is only applied to the default key*

MQTT Map reports which are not encrypted by the well-known PSK or any other PSK (`decoded` in protobufspeak) are not filtered by the meshtasic MQTT (as far as I can tell). They are available with precise locations and with a precision of less than 16 bits.

With the right map and MQTT topics, accurate positions are available Meshmap.net. It subscribes to topics with the form `msh/"LORA_REGION"/#` on the  meshtasic MQTT and maps the locations of position reports (portnum 3) and map reports (portnum 73).

For SAR purposes, mentioned in @cyberorg's comment, assuming availability of a network connection for MQTT is does not suitable for all situations but it could be an alternative to getting other users to configure another channel that does not use the well-known PSK.

**Comment by JaraLowell** [comment]
So far from tests we seen that this basically null's travelling nodes on road if they have precision a bit higher so they actually show to be on a road. As moment the position data or map reporting is ignored any other package is not relayed either or least that what we see so far. one can trace a node, one can chat to a node, one can request info; but beacons to ^all seem to get blocked moment your map info was not within that bit set.

Not sure this gone aid for new people that are like why am i not visible, if within the Mestastic app there be no warning or the likes explaining this. Wish i think should had been the route to go and not filtering it on mqtt side. Simply agree button in the app stating you set precision so your within a meter can see where your node is, do you agree prolly would bin better. Seeing most igates already changed where there node was or is.

Not every one in end reads these posts and thus have no clue why it for some reason now does not work no more.

Anyways that be my two cent's. I do understand per certain laws the privacy might be an issue; yes, but that same issue be on APRS to or any radio tracking website. Creating awareness be in that more important. If i wanted to track a specific user i prolly would not use Meshtastic anyways.. as far as i know,. apple has these wee lil thingies you can drop in a car that do a far better job and not seems blocked that either.

**Reply by valzzu** [reply]
364 m precision is the lowest that will get sent to mqtt
